### PR TITLE
Added modulus/remainder note

### LINKED ISSUE #4884: VB: Modulus operator should be referred to as Remainder
See #4860 

The operator referred to as "modulus" performs a "remainder" operation. There are subtle differences, primarily with negative numbers.

As pointed out in the PR, the `Mod` operator name complicates this.  (Similarly `op_Modulus` in IL complicates the explanation as well.

### PR DESCRIPTION
**Pull request by rpetrusha**
## Added modulus/remainder note

The PR:
- Notes that Mod performs a remainder operation.
- Changes headers to sentence case.
- Makes some formatting fixes. 

Fixes #4884 

@BillWagner @KathleenDollard 

//cc @aaronfranke @pkulikov

### CODE REVIEW

File: docs/visual-basic/language-reference/operators/mod-operator.md [hunk 1]
```diff
@@ -38,12 +35,34 @@ number1 Mod number2
...
- The result is the remainder after `number1` is divided by `number2`. For example, the expression `14 Mod 4` evaluates to 2.  
-  
+## Result
+
+The result is the remainder after `number1` is divided by `number2`. For example, the expression `14 Mod 4` evaluates to 2.  
+
+> [!NOTE]
+> There is a difference between *remainder* and *modulus* in mathematics, with different results for negative numbers. The `Mod` operator in Visual Basic, the .NET Framework op_Modulus operator, and the underlying [rem]<xref:System.Reflection.Emit.OpCodes.Rem> IL instruction all perform a remainder operation.
+
+The result of a `Mod` operation is always in the range (`number2`, -`number2`), exclusive, and retains the sign of the dividend, (`number1`). For example:
```
Comment (KathleenDollard) [inline docs/visual-basic/language-reference/operators/mod-operator.md]: Could we say this more simply as (assuming we use the terms dividend and divisor frequently in the docs). 

The result of a `Mod` operation retains the sign of the dividend, (`number`1), so may be positive or negative. If the dividend is positive, the result is always between zero and the divisor (`number 2`). If the dividend is negative, the result is always between zero and the negative of the divisor (`-number 2`). For example:

Comment (pkulikov) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: The range (`number2`, `-number2`) reads unusual for negative numbers, e.g., (3, -3). I like the version suggested by @KathleenDollard but it's doesn't work when `number2` is negative. What about the following edition (I might be wrong with the `Abs` syntax):

The result of a `Mod` operation retains the sign of the dividend, (`number1`), so may be positive or negative. If the dividend is positive, the result is always between zero and the absolute value of the divisor (`Abs(number2)`). If the dividend is negative, the result is always between the negative of the absolute value of the divisor (`-Abs(number2)`) and zero. For example:

Comment (pkulikov) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: OR:

The result of a `Mod` operation retains the sign of the dividend, `number1`, so may be positive or negative. The absolute value of the result is less than the absolute value of the divisor, `Abs(number2)`

Comment (BillWagner) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: Another way is:

The result of a Mod operation retains the sign of the dividend, `number1`, so may be positive or negative. The result is always between 0 and `number2`.

**Review by BillWagner — APPROVED** [review]
This LGTM @rpetrusha You can merge once you update per other comments.

File: docs/visual-basic/language-reference/operators/mod-operator.md (hunk 1 above)
Comment (pkulikov) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: Consider number1 = -8, number2 = 3. Then result is -2 and not between 0 and 3. I like that @rpetrusha shows all possible sign combinations in the example. Maybe it's better than all the words :)

**Review by aaronfranke — APPROVED** [review]
Some minor problems, otherwise great!

File: docs/visual-basic/language-reference/operators/mod-operator.md (hunk 1 above)
Comment (KathleenDollard) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: I like your correction to mine, but agree that the point is to support the great examples. I wonder if calling out ignoring the divisor makes sense. 

The result of a Mod operation retains the sign of the dividend, (number1), so may be positive or negative. The sign of the divisor is ignored and the absolute value is used, (Abs(number2)).  If the dividend is positive, the result is always between zero and the absolute value of the divisor (Abs(number2)). If the dividend is negative, the result is always between the negative of the absolute value of the divisor (-Abs(number2)) and zero. For example:

I'm fine with any of them, but found I had to do mental gymnastics to figure out the range explanation, even if it is more concise. If we can be verbose here, we could include both as they will speak to different people.

File: docs/visual-basic/language-reference/operators/mod-operator.md [hunk 2]
```diff
@@ -55,17 +74,17 @@ number1 Mod number2
...
+## Equivalent formula  
  The expression `a Mod b` is equivalent to either of the following formulas:  
   
  `a - (b * (a \ b))`  
   
  `a - (b * Fix(a / b))`  
   
-## Floating-Point Imprecision  
+## Floating-point imprecision  
  When you work with floating-point numbers, remember that they do not always have a precise representation in memory. This could lead to unexpected results from certain operations, such as value comparison and the `Mod` operator. For more information, see [Troubleshooting Data Types](../../../visual-basic/programming-guide/language-features/data-types/troubleshooting-data-types.md).
```
[the remark this answers is not in the document]
Comment (KathleenDollard) [reply docs/visual-basic/language-reference/operators/mod-operator.md]: Well, technically it is precise but not accurate <gdr>

**Pull request merged**